robots/norbot2.h

```diff
@@ -0,0 +1,99 @@
+
+#pragma once
+
+// BoB robotics includes
+#include "../common/i2c_interface.h"
+
+
+// Standard C includes
+#include <cmath>
+#include <cstdint>
+
+// Standard C++ includes
+#include <vector>
+
+// third party includes
+#include "../third_party/units.h"
+
+
+//----------------------------------------------------------------------------
+
+//----------------------------------------------------------------------------
+//! An interface for 4 wheeled, Arduino-based robots developed at the University of Sussex
+class Norbot2 
+{
+    using meters_per_second_t = units::velocity::meters_per_second_t;
+    using millimeter_t = units::length::millimeter_t;
+    
+
+public:
+    Norbot2(const char *path = "/dev/i2c-1", int slaveAddress = 0x29)
+      : m_I2C(path, slaveAddress)
+      , m_forwardVelocity(0.0f)
+      , m_turningAngle(0.0f)
+    {}
+
+    // destructor		
+    ~Norbot2() 
+    {
+        stopMoving();
+    }
+
+    void move(uint8_t speed, uint8_t turningAngle)
+    {
+      
+        // Cache left and right
+        m_forwardVelocity = speed;      // 0-127 backward, 128-255 forward
+        m_turningAngle = turningAngle;  // 55 left, 90 center, 125 right
```

Comment:
I'd have the range of ``turningAngle`` be from -45° to 45° or whatever, so that 0° is the centre.